### PR TITLE
(feat) O3-1165 Register Anonymous Patients

### PR DESCRIPTION
**Pull request by ZacButko**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ x I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
Allows to register an anonymous patient. There is a new Content Switcher asking if the patient's name is known. If it is, then proceed as normal. If not then patient name fields are removed, and hidden input fields supply values for givenName and familyName. These defaults are specified in the config schema under fieldConfigurations.

## Screenshots
anonymous-patient

## Related Issue
[O3-1165](https://issues.openmrs.org/browse/O3-1165)

## Other
Cleaned up validation for gender and dob as well, and unified content switcher carbon styling

### CODE REVIEW

**Comment by rbuisson** [comment]
Thanks for this PR @ZacButko . In the Ref App 2.x, there is a more discreet button to set a patient as anonymous.

Screenshot 2022-03-25 at 15 05 02

In most cases the patient name will be known, and in only few the patient name will be unknown.
Is there a way to make this less invasive for the user by default?

As in your screenshot, that's quite taking some space and that is the really the first thing that the user will come accross.

File: packages/esm-patient-registration-app/src/patient-registration/field/name/name-field.component.tsx [hunk 1]
```diff
@@ -41,29 +61,57 @@ export const NameField = () => {
...
-              checkWarning={checkNumber}
-            />
+          <div className={styles.dobContentSwitcherLabel}>
+            <span className={styles.label01}>{t('patientNameKnown', "Patient's Name is Known?")}</span>
+          </div>
+          <ContentSwitcher onChange={toggleNameKnown}>
+            <Switch name="known" text={t('yes', 'Yes')} />
+            <Switch name="unknown" text={t('no', 'No')} />
+          </ContentSwitcher>
+          <div style={{ minHeight: '1rem' }}></div>
```
[the remark this answers is not in the document]
Comment (ZacButko) [reply packages/esm-patient-registration-app/src/patient-registration/field/name/name-field.component.tsx:71]: Actually it's worse than that. This is just ad-hock spacing between form elements. The better option here is to have this as a part of the form grid and have grid spacing clearly defined

File: packages/esm-patient-registration-app/translations/en.json [hunk 2]
```diff
@@ -48,9 +47,11 @@
   "months": "Months",
   "negativeMonths": "Months cannot be less than 0",
   "negativeYears": "Years cannot be less than 0",
+  "no": "No",
   "numberInNameDubious": "Name should not contain numbers",
   "otherLabelText": "Other",
   "patient": "Patient",
+  "patientNameKnown": "Patient's Name is Known?",
```
Comment (ibacher) [inline packages/esm-patient-registration-app/translations/en.json:54]:
```suggestion
  "patientNameKnown": "Is Patient's Name Known?",
```

**Review by ibacher — APPROVED** [review]
LGTM!

**Pull request merged**